### PR TITLE
USHIFT-2970: RHEL 9.4 PreSubmits

### PR DESCRIPTION
**Pull request by copejon**
[USHIFT-2970](https://issues.redhat.com//browse/USHIFT-2970)

### CODE REVIEW

File: test/image-blueprints/layer2-presubmit/group1/rhel94-crel.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  around the first and last template instructions.
+
+*/ -}}
+
+name = "rhel-9.4-microshift-crel"
+description = "A RHEL 9.4 image with already built and released RPMs like EC, RC, or Z-stream release"
+version = "0.0.1"
+modules = []
+groups = []
+distro = "rhel-92"
```
Comment (ggiguash) [inline test/image-blueprints/layer2-presubmit/group1/rhel94-crel.toml:16]: Wrong distro

File: test/image-blueprints/layer2-presubmit/group1/rhel94-source-base.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name = "rhel-9.2-microshift-source-base"
+description = "A RHEL 9.2 image with the RPMs built from source from the base release branch"
+version = "0.0.1"
+modules = []
+groups = []
+distro = "rhel-92"
```
Comment (ggiguash) [inline test/image-blueprints/layer2-presubmit/group1/rhel94-source-base.toml:6]: Wrong distro

File: test/image-blueprints/layer2-presubmit/group1/rhel94-source.toml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+name = "rhel-9.2-microshift-source"
+description = "A RHEL 9.2 image with the RPMs built from source."
+version = "0.0.1"
+modules = []
+groups = []
+distro = "rhel-92"
```
Comment (ggiguash) [inline test/image-blueprints/layer2-presubmit/group1/rhel94-source.toml:6]: Wrong distro

File: test/image-blueprints/layer2-presubmit/group1/rhel94-source-base.toml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+name = "rhel-9.2-microshift-source-base"
```
Comment (ggiguash) [inline test/image-blueprints/layer2-presubmit/group1/rhel94-source-base.toml:1]: Like in my previous comment, naming convention is a bit confusing.
Should this be `rhel-9.4`?

Comment (ggiguash) [reply test/image-blueprints/layer2-presubmit/group1/rhel94-source-base.toml:1]: In general, file name should correspond to what's built inside.

File: test/image-blueprints/layer2-presubmit/group1/rhel94-source.toml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+name = "rhel-9.2-microshift-source"
```
Comment (ggiguash) [inline test/image-blueprints/layer2-presubmit/group1/rhel94-source.toml:1]: Inconsistent naming with the file.

File: go.sum [hunk 6]
```diff
@@ -56,7 +56,7 @@ github.com/Azure/go-autorest v14.2.0+incompatible h1:V5VMDjClD3GiElqLWO7mz2MxNAK
 github.com/Azure/go-autorest v14.2.0+incompatible/go.mod h1:r+4oMnoxhatjLLJ6zxSWATqVooLgysK6ZNox3g/xq24=
 github.com/Azure/go-autorest/autorest v0.11.29 h1:I4+HL/JDvErx2LjyzaVxllw2lRDB5/BT2Bm4g20iqYw=
 github.com/Azure/go-autorest/autorest v0.11.29/go.mod h1:ZtEzC4Jy2JDrZLxvWs8LrBWEBycl1hbT1eknI8MtfAs=
-github.com/Azure/go-autorest/autorest/adal v0.9.22/go.mod h1:XuAbAEUv2Tta//+voMI038TrJBqjKam0me7qR+L8Cmk=
+github.com/Azure/go-autorest/autorest/adal v0.9.42/go.mod h1:XuAbAEUv2Tta//+voMI038TrJBqjKam0me7qR+L8Cmk=
```
Comment (pmtk) [inline go.sum:59]: I don't think this should happen, right?

File: test/image-blueprints/layer2-presubmit/group1/rhel92-source-base.toml [hunk 7]
```diff
@@ -1,9 +1,9 @@
 name = "rhel-9.2-microshift-source-base"
-description = "A RHEL 9.2 image with the RPMs built from source from the base release branch"
+description = "A RHEL 9.4 image with the RPMs built from source from the base release branch"
 version = "0.0.1"
 modules = []
 groups = []
-distro = "rhel-92"
+distro = "rhel-94"
```
Comment (pmtk) [inline test/image-blueprints/layer2-presubmit/group1/rhel92-source-base.toml:6]: Mismatch between `distro`, `name`, and filename.

File: test/scenarios/el94-base@upgrade-ok.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (pmtk) [inline test/scenarios/el94-base@upgrade-ok.sh:1]: I don't see any removals in this dir, shouldn't we remove scenarios on el92 and el93 unless they upgrade to el94?

File: test/scenarios/el94-base@el94-src@upgrade-ok.sh [hunk 9]
```diff
@@ -13,6 +13,5 @@ scenario_remove_vms() {
 
 scenario_run_tests() {
     run_tests host1 \
-        --variable "TARGET_REF:rhel-9.2-microshift-source" \
```
Comment (pacevedom) [inline test/scenarios/el94-base@el94-src@upgrade-ok.sh:16]: Need to update this one to `rhel-9.4-microshift-source`?

File: go.sum (hunk 6 above)
Comment (copejon) [reply go.sum]: Don't know why that occurred, fixed now

File: test/image-blueprints/layer1-base/group2/rhel94-microshift-previous-minor.toml [hunk 10]
```diff
@@ -0,0 +1,39 @@
+name = "rhel-9.3-microshift-4.{{ .Env.PREVIOUS_MINOR_VERSION }}"
```
[the remark this answers is not in the document]
Comment (copejon) [reply test/image-blueprints/layer1-base/group2/rhel94-microshift-previous-minor.toml]: The description was wrong for the distro, corrected the description and name to use 94

File: test/scenarios/el94-base@el94-src@upgrade-ok.sh (hunk 9 above)
Comment (copejon) [reply test/scenarios/el94-base@el94-src@upgrade-ok.sh:16]: i've restored the truncated argument and corrected the version to 9.4

File: test/scenarios/el94-crel@el94-src@upgrade-ok.sh [hunk 11]
```diff
@@ -27,6 +27,6 @@ scenario_run_tests() {
         return 0
     fi
     run_tests host1 \
-              --variable "TARGET_REF:rhel-9.2-microshift-source" \
-              suites/upgrade/upgrade-successful.robot
+              suites/upgrade/upgrade-successful.robot \
+              --variable "TARGET_REF:rhel-9.4-microshift-source"
```
Comment (pacevedom) [inline test/scenarios/el94-crel@el94-src@upgrade-ok.sh:31]: Need to swap the order here, `--variable ...` is interpreted as a file too.

File: test/image-blueprints/layer2-presubmit/group1/rhel92-source-base.toml [hunk 12]
```diff
@@ -1,5 +1,5 @@
 name = "rhel-9.2-microshift-source-base"
-description = "A RHEL 9.2 image with the RPMs built from source from the base release branch"
+description = "A RHEL 9.4 image with the RPMs built from source from the base release branch"
```
Comment (pmtk) [inline test/image-blueprints/layer2-presubmit/group1/rhel92-source-base.toml:2]:
```suggestion
description = "A RHEL 9.2 image with the RPMs built from source from the base release branch"
```

**Pull request merged**